kosat-api/src/commonTest/kotlin/KotlinApiTest.kt

```diff
@@ -0,0 +1,86 @@
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class KotlinApiTest {
+    @Test
+    fun tieShirtVarargs() {
+        val solver = Kosat()
+
+        assertFailsWith(IllegalArgumentException::class) { solver.newClause(1, 2) }
```

Comment:
What about `assertFailsWith<IllegalStateException> { ... }` ? No need for `::class` when we have `reified` :)